newm-server/src/main/kotlin/io/newm/server/features/marketplace/daemon/MarketplaceMonitorDaemon.kt

```diff
@@ -0,0 +1,96 @@
+package io.newm.server.features.marketplace.daemon
+
+import io.ktor.server.application.ApplicationEnvironment
+import io.newm.chain.grpc.MonitorAddressResponse
+import io.newm.server.features.cardano.repo.CardanoRepository
+import io.newm.server.features.marketplace.repo.MarketplaceRepository
+import io.newm.server.ktx.getSecureConfigString
+import io.newm.shared.daemon.Daemon
+import io.newm.shared.koin.inject
+import io.newm.shared.ktx.error
+import io.newm.shared.ktx.getConfigBoolean
+import io.newm.shared.ktx.getConfigLong
+import io.newm.shared.ktx.info
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.cancelChildren
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import org.apache.curator.framework.recipes.leader.LeaderLatch
+import org.apache.curator.framework.recipes.leader.LeaderLatchListener
+import org.koin.core.parameter.parametersOf
+import org.slf4j.Logger
+import kotlin.time.Duration.Companion.seconds
+
+private const val LEADER_LATCH_PATH = "/leader-latches/marketplace-monitor"
+
+class MarketplaceMonitorDaemon(
```

Comment:
This looked like it was fun to write.

Reply:
Yes, definitely. This one was a fun one. It took me several iterations to get it to a pattern that I like.